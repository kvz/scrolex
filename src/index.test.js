require('babel-polyfill')
const Scrolex = require('./index')
// const debug   = require('depurar')('sut')

describe('index', () => {
  describe('exe', () => {
    it('should accept callback', () => {
      Scrolex.exe(`MOCK_ERROR_OUT=0 MOCK_LIMIT=2 node ${__dirname}/fakecmd.js`, { cleanupTmpFiles: false, mode: 'passthru', components: 'lanyon>postinstall' }, (err, out) => {
        expect(err).toBeNull()
        expect(out).toMatchSnapshot()
      })
    })
    it('should accept promise and catch an error', () => {
      Scrolex.exe(`MOCK_ERROR_OUT=1 MOCK_LIMIT=1 node ${__dirname}/fakecmd.js`, { cleanupTmpFiles: false, mode: 'singlescroll', components: 'lanyon>postinstall' })
        .then(() => {
          expect(true).toBe(false)
        })
        .catch(({message}) => {
          expect(message).toMatchSnapshot()
        })
    })
    it('should accept async/await', async () => {
      let out = await Scrolex.exe(`MOCK_ERROR_OUT=0 MOCK_LIMIT=2 node ${__dirname}/fakecmd.js`, { cleanupTmpFiles: false, mode: 'singlescroll', components: 'lanyon>postinstall' })
      expect(out).toMatchSnapshot()
    })
    it('should accept async/await and catch an error', () => {
      expect(async () => {
        await Scrolex.exe(`MOCK_ERROR_OUT=1 MOCK_LIMIT=1 node ${__dirname}/fakecmd.js`, { cleanupTmpFiles: false, mode: 'singlescroll', components: 'lanyon>postinstall' })
      }).toThrowErrorMatchingSnapshot
    })
  })
})
