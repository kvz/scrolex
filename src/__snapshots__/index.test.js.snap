exports[`index exe should accept async/await 1`] = `
"

    Starting with limit=1, interval=1000, errorOut=true
    Doing thing 1. fakecmd with limit=1, interval=1000, errorOut=true done.

. Fault while executing \"MOCK_ERROR_OUT=1 MOCK_LIMIT=1 node #{SCROLEX_ROOT}/src/fakecmd.js\". Exit code: 1"
`;

exports[`index exe should accept async/await 2`] = `
"Starting with limit=2, interval=1000, errorOut=false
Doing thing 1
Doing thing 2. fakecmd with limit=2, interval=1000, errorOut=false done."
`;

exports[`index exe should accept async/await 3`] = `
"Starting with limit=2, interval=1000, errorOut=false
Doing thing 1
Doing thing 2. fakecmd with limit=2, interval=1000, errorOut=false done."
`;

exports[`index exe should accept async/await 4`] = `
"Starting with limit=2, interval=1000, errorOut=false
Doing thing 1
Doing thing 2. fakecmd with limit=2, interval=1000, errorOut=false done."
`;

exports[`index exe should accept async/await and catch an error 1`] = `
[Error: 

    Starting with limit=1, interval=1000, errorOut=true
    Doing thing 1. fakecmd with limit=1, interval=1000, errorOut=true done.

. Fault while executing "MOCK_ERROR_OUT=1 MOCK_LIMIT=1 node #{SCROLEX_ROOT}/src/fakecmd.js". Exit code: 1]
`;
